File: src/lib/api/matching.ts
import { apiClient } from './client';
import { Vessel } from './vessels';

export interface MatchingRuleResult {
  rule_name: string;
  rule_category: string;
  status: 'RELEVANT' | 'PARTIAL' | 'NOT_RELEVANT' | 'UNKNOWN';
  evaluated_value?: string;
  required_value?: string;
  explanation: string;
  is_blocking: boolean;
}

export interface MatchingCandidate {
  id: string;
  vessel_id: string;
  vessel?: Vessel;
  overall_status: 'RELEVANT' | 'PARTIAL' | 'NOT_RELEVANT' | 'UNKNOWN';
  summary_explanation: string;
  is_data_complete: boolean;
  missing_fields?: string;
  rule_results: MatchingRuleResult[];
}

export interface MatchingRun {
  id: string;
  requirement_id: string;
  run_timestamp: string;
  total_evaluated: number;
  relevant_count: number;
  partial_count: number;
  not_relevant_count: number;
  unknown_count: number;
  candidates: MatchingCandidate[];
}

export const matchingApi = {
  evaluate: (requirementId: string, candidateVesselIds?: string[]) =>
    apiClient<MatchingRun>('/matching/evaluate', {
      method: 'POST',
      body: JSON.stringify({
        requirement_id: requirementId,
        candidate_vessel_ids: candidateVesselIds,
      }),
    }),
  getLatestForRequirement: (requirementId: string) =>
    apiClient<MatchingRun>(`/matching/requirements/${requirementId}/latest`),
};

File: backend/app/schemas/matching.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional
from datetime import datetime
from app.models.enums import MatchStatus
from app.schemas.vessels import VesselResponse

class MatchingRuleResultResponse(BaseModel):
    rule_name: str
    rule_category: str
    status: MatchStatus
    evaluated_value: Optional[str] = None
    required_value: Optional[str] = None
    explanation: str
    is_blocking: bool = False

    model_config = ConfigDict(from_attributes=True)

class MatchingCandidateResponse(BaseModel):
    id: str
    vessel_id: str
    vessel: Optional[VesselResponse] = None
    overall_status: MatchStatus
    summary_explanation: str
    is_data_complete: bool
    missing_fields: Optional[str] = None
    rule_results: List[MatchingRuleResultResponse] = []

    model_config = ConfigDict(from_attributes=True)

class MatchingRunResponse(BaseModel):
    id: str
    requirement_id: str
    run_timestamp: datetime
    total_evaluated: int
    relevant_count: int
    partial_count: int
    not_relevant_count: int
    unknown_count: int
    candidates: List[MatchingCandidateResponse] = []

    model_config = ConfigDict(from_attributes=True)

class MatchingRunRequest(BaseModel):
    requirement_id: str
    candidate_vessel_ids: Optional[List[str]] = None
